File: js/chiosco.js
/**
 * Chiosco - Macchina a Stati Finiti (FSM)
 * Coordina tutti i componenti e gestisce le transizioni di stato
 *
 * Stati:
 * - IDLE: Stato iniziale, in attesa di azione utente
 * - PAGAMENTO_MONETE: Inserimento monete in corso
 * - PAGAMENTO_CARTA: Pagamento con carta di credito VISA in corso
 * - VERIFICA_QR: Verifica codice QR autorizzato in corso
 * - PORTA_APERTA: Porta aperta dopo pagamento/autorizzazione
 * - TIMEOUT: Timeout inattività durante operazione
 * - MANUTENZIONE_AUTH_PENDING: Attesa autenticazione operatore (10s countdown)
 * - MANUTENZIONE_ATTESA_CHIUSURA: Operatore autenticato, attesa chiusura cassetta
 * - MANUTENZIONE_SCELTA_AZZERAMENTO: Scelta azzeramento saldo monete
 * - FUORI_SERVIZIO: Sistema non disponibile (timeout auth), richiede reset operatore
 */

class Chiosco {
  constructor() {
    this.stato = 'IDLE';

    // Transizioni permesse
    this.transizioniPermesse = {
      'IDLE': ['PAGAMENTO_MONETE', 'PAGAMENTO_CARTA', 'VERIFICA_QR', 'PORTA_APERTA', 'MANUTENZIONE_AUTH_PENDING'],
      'PAGAMENTO_MONETE': ['PORTA_APERTA', 'TIMEOUT', 'IDLE', 'MANUTENZIONE_AUTH_PENDING'],
      'PAGAMENTO_CARTA': ['PORTA_APERTA', 'IDLE'],
      'VERIFICA_QR': ['PORTA_APERTA', 'IDLE'],
      'PORTA_APERTA': ['IDLE', 'MANUTENZIONE_AUTH_PENDING'],
      'TIMEOUT': ['IDLE'],
      'MANUTENZIONE_AUTH_PENDING': ['MANUTENZIONE_ATTESA_CHIUSURA', 'FUORI_SERVIZIO', 'IDLE'],
      'MANUTENZIONE_ATTESA_CHIUSURA': ['MANUTENZIONE_SCELTA_AZZERAMENTO', 'FUORI_SERVIZIO'],
      'MANUTENZIONE_SCELTA_AZZERAMENTO': ['IDLE'],
      'FUORI_SERVIZIO': ['IDLE']
    };

    // Riferimenti ai componenti (saranno impostati da app.js)
    this.display = null;
    this.porta = null;
    this.gettoniera = null;
    this.lettoreCarte = null;
    this.lettoreQR = null;
    this.gestoreTimeout = null;

    // Feature 003: Componenti manutenzione cassetta
    this.sensoreCassetta = new SensoreCassetta();
    this.suoneria = new Suoneria();
    this.gestoreManutenzione = new GestoreManutenzione(this);
    this.operazioneCorrente = null;
    this.pendenteAperturaCassetta = false;

    // Feature 004: Registra listener per eventi cassetta (nuovo pattern)
    this.sensoreCassetta.on('cambioStato', (dati) => {
      if (dati.statoNuovo === 'APERTA') {
        this.onCassettaAperta();
      } else if (dati.statoNuovo === 'CHIUSA') {
        this.onCassettaChiusa();
      }
    });

    // Mappa degli stati (State Pattern)
    this.stati = {
      'IDLE': new StatoIdle(),
      'PAGAMENTO_MONETE': new StatoPagamentoMonete(),
      'PAGAMENTO_CARTA': new StatoPagamentoCarta(),
      'VERIFICA_QR': new StatoVerificaQR(),
      'PORTA_APERTA': new StatoPortaAperta(),
      'TIMEOUT': new StatoTimeout(),
      'MANUTENZIONE_AUTH_PENDING': new StatoManutenzioneAuthPending(),
      'MANUTENZIONE_ATTESA_CHIUSURA': new StatoManutenzioneAttesaChiusura(),
      'MANUTENZIONE_SCELTA_AZZERAMENTO': new StatoManutenzioneSceltaAzzeramento(),
      'FUORI_SERVIZIO': new StatoFuoriServizio()
    };

    this.statoCorrente = this.stati['IDLE'];

    log.info('🏗️ Chiosco inizializzato - Stato: IDLE');
  }

  /**
   * Tenta una transizione di stato
   * @param {string} nuovoStatoNome - Nome dello stato target
   * @param {object} dati - Dati opzionali per la transizione
   * @returns {boolean} true se transizione riuscita
   */
  transizione(nuovoStatoNome, dati = {}) {
    // Verifica se transizione è permessa
    if (!this.transizioniPermesse[this.stato]?.includes(nuovoStatoNome)) {
      log.error(`❌ Transizione non permessa: ${this.stato} → ${nuovoStatoNome}`);
      return false;
    }

    const nuovoStato = this.stati[nuovoStatoNome];
    if (!nuovoStato) {
      log.error(`❌ Stato non trovato: ${nuovoStatoNome}`);
      return false;
    }

    const vecchioStatoNome = this.stato;

    // Esegui uscita stato precedente (opzionale)
    this.statoCorrente.esci(this);

    // Aggiorna stato corrente
    this.stato = nuovoStatoNome; // Mantiene compatibilità stringa
    this.statoCorrente = nuovoStato;

    log.info(`🔄 Transizione: ${vecchioStatoNome} → ${nuovoStatoNome}`);

    // Esegui entrata nuovo stato
    this.statoCorrente.entra(this, dati);

    return true;
  }

  /**
   * Abilita/disabilita input
   * Delega a Display
   */
  abilitaInput(abilitato, eccezioni = []) {
    if (this.display) {
      this.display.abilitaInput(abilitato, eccezioni);
    }
  }

  /**
   * Verifica pagamento completato
   */
  verificaPagamento() {
    if (this.stato !== 'PAGAMENTO_MONETE') {
      return;
    }

    const rimanente = this.gettoniera.getImportoRimanente();

    if (rimanente <= 0) {
      log.info('✅ Pagamento completato');
      this.transizione('PORTA_APERTA', { motivo: 'monete' });
    }
  }

  /**
   * FEATURE 002: Handler passaggio persona
   * Chiamato quando utente clicca pulsante "Persona passata"
   */
  onPassaggioPersona() {
    if (this.stato !== 'PORTA_APERTA') {
      log.warn('⚠️ Tentativo passaggio persona con porta non aperta - Ignorato');
      return;
    }

    log.debug('🚶 Click pulsante "Persona passata"');

    // Disabilita e nascondi pulsante immediatamente
    if (this.display) {
      this.display.gestisciPulsantePassaggio(false, false);
    }

    // Mostra feedback su display
    if (this.display) {
      this.display.mostraMessaggio('Passaggio rilevato - Porta in chiusura', 'info');
    }

    // Chiusura manuale porta (cancella timer e chiude)
    if (this.porta) {
      this.porta.chiudiManuale();
    }

    // Torna a IDLE dopo animazione chiusura (1.5s)
    setTimeout(() => {
      this.transizione('IDLE');
    }, 1500);
  }

  /**
   * FEATURE 003: Handler apertura cassetta
   */
  onCassettaAperta() {
    const statiTransito = ['PORTA_APERTA', 'PAGAMENTO_MONETE'];

    if (statiTransito.includes(this.stato)) {
      log.info('Cassetta aperta durante transito - attesa fine operazione');
      this.pendenteAperturaCassetta = true;

      // Se pagamento monete, annulla
      if (this.stato === 'PAGAMENTO_MONETE') {
        log.warn('Pagamento monete annullato per manutenzione');
        if (this.gettoniera) {
          this.gettoniera.reset();
        }
        this.transizione('MANUTENZIONE_AUTH_PENDING');
      }
    } else if (this.stato === 'IDLE') {
      this.transizione('MANUTENZIONE_AUTH_PENDING');
    } else {
      log.warn('Apertura cassetta ignorata, stato incompatibile: ' + this.stato);
    }
  }

  /**
   * FEATURE 003: Handler chiusura cassetta
   */
  onCassettaChiusa() {
    if (this.stato === 'MANUTENZIONE_ATTESA_CHIUSURA') {
      if (this.operazioneCorrente) {
        this.operazioneCorrente.logEvento('CHIUSURA');
      }
      this.transizione('MANUTENZIONE_SCELTA_AZZERAMENTO');
    }
  }

  /**
   * FEATURE 003: Handler conferma azzeramento (T032)
   * @param {boolean} scelta - true per azzerare, false per mantenere
   */
  confermaAzzeramento(scelta) {
    if (this.stato !== 'MANUTENZIONE_SCELTA_AZZERAMENTO') {
      log.warn('⚠️ Tentativo conferma azzeramento fuori contesto - Ignorato');
      return;
    }

    if (scelta) {
      // Azzera saldo
      const saldoPrecedente = this.gettoniera ? this.gettoniera.azzeraSaldo() : 0;

      if (this.operazioneCorrente) {
        this.operazioneCorrente.logEvento('AZZERAMENTO', {
          azzerato: true,
          saldoPrima: saldoPrecedente,
          saldoDopo: 0
        });
      }

      if (this.display) {
        this.display.nascondiPulsantiAzzeramento();
        this.display.mostraMessaggio(`Saldo azzerato: ${Gettoniera.formattaImporto(saldoPrecedente)}`, 'successo');
      }

      log.info(`✅ Saldo azzerato: ${Gettoniera.formattaImporto(saldoPrecedente)}`);
    } else {
      // Mantieni saldo
      const saldoCorrente = this.gettoniera ? this.gettoniera.getSaldoMonete() : 0;

      if (this.operazioneCorrente) {
        this.operazioneCorrente.logEvento('AZZERAMENTO', {
          azzerato: false,
          saldoPrima: saldoCorrente,
          saldoDopo: saldoCorrente
        });
      }

      if (this.display) {
        this.display.nascondiPulsantiAzzeramento();
        this.display.mostraMessaggio(`Saldo mantenuto: ${Gettoniera.formattaImporto(saldoCorrente)}`, 'info');
      }

      log.info(`ℹ️ Saldo mantenuto: ${Gettoniera.formattaImporto(saldoCorrente)}`);
    }

    // Chiudi operazione manutenzione
    this.operazioneCorrente = null;

    // Torna a IDLE dopo 3 secondi
    setTimeout(() => {
      this.transizione('IDLE');
    }, 3000);
  }

  /**
   * FEATURE 003: Reset da FUORI_SERVIZIO con carta autorizzata
   */
  resetDaFuoriServizio(codiceOperatore) {
    if (!Validatore.isCodiceAutorizzato(codiceOperatore)) {
      if (this.display) {
        this.display.mostraMessaggio(`Accesso negato (${codiceOperatore})`, 'error');
      }
      setTimeout(() => {
        if (this.display) {
          this.display.mostraFuoriServizio();
        }
      }, 2000);
      return;
    }

    // Reset valido
    const operazione = new OperazioneSvuotamento();
    operazione.logEvento('RESET', { codice: codiceOperatore });

    this.suoneria.disattiva();

    if (this.display) {
      this.display.mostraMessaggio(`Sistema ripristinato da operatore (${codiceOperatore})`, 'success');
    }

    setTimeout(() => {
      this.transizione('IDLE');
    }, 3000);

    log.info(`✅ Reset da FUORI_SERVIZIO - Operatore: ${codiceOperatore}`);
  }

  /**
   * Helper per verifica accesso con codice autorizzato (DRY helper)
   * Centralizza logica comune per QR e carte autorizzate
   *
   * @param {string} codice - Codice da verificare
   * @param {string} tipoIngresso - Tipo ingresso ('QR', 'Carta', ecc.) per logging
   */
  verificaAccessoConCodice(codice, tipoIngresso) {
    if (!this.display) {
      log.warn('verificaAccessoConCodice: display non disponibile');
      return;
    }

    this.display.mostraMessaggio('Verifica in corso...', 'info');

    setTimeout(() => {
      const autorizzato = Validatore.isCodiceAutorizzato(codice);

      if (autorizzato) {
        log.info(`✅ ${tipoIngresso} ${codice}: AUTORIZZATO`);
        this.display.mostraMessaggio('Accesso autorizzato', 'successo');
        setTimeout(() => this.transizione('PORTA_APERTA', { motivo: tipoIngresso.toLowerCase() }), 1000);
      } else {
        log.warn(`⚠️ ${tipoIngresso} ${codice}: NON AUTORIZZATO`);
        this.display.mostraMessaggio('Accesso negato', 'errore');
        setTimeout(() => this.transizione('IDLE'), 2000);
      }
    }, 500);
  }

  /**
   * Gestisce input da lettore carte
   * Delega logica allo stato corrente (State Pattern)
   * @param {string} codice - Codice carta letto
   */
  verificaCarta(codice) {
    this.statoCorrente.gestisciInputCarta(this, codice);
  }

  /**
   * Reset completo del chiosco (per debugging)
   */
  reset() {
    log.warn('🔄 Reset manuale chiosco');
    if (this.suoneria?.isAttiva()) {
      this.suoneria.disattiva();
    }
    if (this.gestoreManutenzione) {
      this.gestoreManutenzione.fermaCountdown();
    }
    this.transizione('IDLE');
  }
}

/**
 * GestoreTimeout - Gestisce timeout inattività con countdown
 *
 * Refactored per usare CountdownTimer (TD-005)
 */
class GestoreTimeout {
  constructor(chiosco, durataSecondi = 20) {
    this.chiosco = chiosco;
    this.durataSecondi = durataSecondi;

    // Usa CountdownTimer con callbacks
    this.countdown = new CountdownTimer(
      durataSecondi,
      // onTick: aggiorna display ogni secondo
      (secondi) => {
        if (this.chiosco.display) {
          this.chiosco.display.aggiornaCountdown(secondi);
        }
        log.debug(`⏱️ Countdown: ${secondi}s rimanenti`);
      },
      // onComplete: transizione a TIMEOUT
      () => {
        log.warn('⏱️ Timeout scaduto');
        this.chiosco.transizione('TIMEOUT');
      }
    );
  }

  /**
   * Avvia timeout
   */
  avvia() {
    log.debug(`⏱️ Timer timeout avviato: ${this.durataSecondi} secondi`);
    this.countdown.avvia();
  }

  /**
   * Reset timeout
   */
  reset() {
    this.countdown.reset();

    // Nascondi countdown su display
    if (this.chiosco?.display) {
      this.chiosco.display.nascondiCountdown();
    }

    log.debug('⏱️ Timer timeout resettato');
  }
}

// Export globale
window.Chiosco = Chiosco;
window.GestoreTimeout = GestoreTimeout;

log.info('✅ Chiosco e GestoreTimeout caricati');

File: ios-app/ChioscoApp/Core/Chiosco.swift
//
//  Chiosco.swift
//  ChioscoApp
//
//  Macchina a Stati Finiti (FSM) principale
//  Coordina tutti i componenti e gestisce le transizioni di stato
//  Tradotto da: js/chiosco.js
//

import Foundation
import Combine

/// Chiosco - FSM principale che coordina tutti i componenti
class Chiosco: ObservableObject {
    // Stato pubblicato per SwiftUI
    @Published var stato: ChioscoStato = .idle

    // Componenti del chiosco (ObservableObject per reattività)
    @Published var display: DisplayModello
    @Published var porta: PortaModello
    @Published var gettoniera: GettonierapModello

    // Gestore timeout
    private var gestoreTimeout: GestoreTimeout?

    init() {
        // Inizializza componenti
        self.display = DisplayModello()
        self.porta = PortaModello()
        self.gettoniera = GettonierapModello()

        // Inizializza gestore timeout (passa riferimento a self dopo init)
        // Nota: gestoreTimeout viene inizializzato in setupGestoreTimeout()

        Logger.info("🏗️ Chiosco inizializzato - Stato: IDLE")

        // Setup gestore timeout dopo init completo
        setupGestoreTimeout()

        // Mostra messaggio iniziale
        display.mostraMessaggioIniziale()
    }

    /// Setup gestore timeout dopo inizializzazione
    private func setupGestoreTimeout() {
        gestoreTimeout = GestoreTimeout(chiosco: self, durataSecondi: 20)
    }

    /// Tenta una transizione di stato
    /// - Parameters:
    ///   - nuovoStato: Lo stato target
    ///   - dati: Dati opzionali per la transizione
    /// - Returns: true se transizione riuscita
    @discardableResult
    func transizione(_ nuovoStato: ChioscoStato, dati: [String: Any] = [:]) -> Bool {
        // Verifica se transizione è permessa
        guard TransizioniPermesse.isPermessa(da: stato, a: nuovoStato) else {
            Logger.error("❌ Transizione non permessa: \(stato.rawValue) → \(nuovoStato.rawValue)")
            return false
        }

        let vecchioStato = stato
        stato = nuovoStato

        Logger.info("🔄 Transizione: \(vecchioStato.rawValue) → \(nuovoStato.rawValue)")

        // Gestisci cambio stato
        onCambioStato(nuovoStato: nuovoStato, vecchioStato: vecchioStato, dati: dati)

        return true
    }

    /// Handler per cambio di stato
    private func onCambioStato(nuovoStato: ChioscoStato, vecchioStato: ChioscoStato, dati: [String: Any]) {
        switch nuovoStato {
        case .idle:
            onEntraIDLE()
        case .pagamentoMonete:
            onEntraPagamentoMonete()
        case .pagamentoCarta:
            onEntraPagamentoCarta()
        case .verificaQR:
            if let codice = dati["codice"] as? String {
                onEntraVerificaQR(codice: codice)
            }
        case .verificaCarta:
            if let codice = dati["codice"] as? String {
                onEntraVerificaCarta(codice: codice)
            }
        case .portaAperta:
            let motivo = dati["motivo"] as? String ?? "pagamento"
            onEntraPortaAperta(motivo: motivo)
        case .timeout:
            onEntraTimeout()
        default:
            // Stati manutenzione non implementati in questa versione base
            break
        }
    }

    // MARK: - Handler Stati

    /// Entra in stato IDLE
    private func onEntraIDLE() {
        // Reset timeout se attivo
        gestoreTimeout?.reset()

        // Reset gettoniera
        gettoniera.reset()

        // Reset display
        display.mostraMessaggioIniziale()

        Logger.info("🏠 Stato IDLE - Sistema pronto")
    }

    /// Entra in stato PAGAMENTO_MONETE
    private func onEntraPagamentoMonete() {
        // Avvia timeout inattività
        gestoreTimeout?.avvia()

        Logger.info("💰 Pagamento monete iniziato")
    }

    /// Entra in stato PAGAMENTO_CARTA
    private func onEntraPagamentoCarta() {
        // Mostra messaggio su display
        display.mostraMessaggio("Avvicina la carta al lettore", tipo: .info)

        Logger.info("💳 Pagamento carta iniziato")
    }

    /// Entra in stato VERIFICA_QR
    private func onEntraVerificaQR(codice: String) {
        // Usa helper DRY per verifica
        verificaAccessoConCodice(codice: codice, tipoIngresso: "QR code")
    }

    /// Entra in stato VERIFICA_CARTA
    private func onEntraVerificaCarta(codice: String) {
        // Usa helper DRY per verifica
        verificaAccessoConCodice(codice: codice, tipoIngresso: "Carta")
    }

    /// Entra in stato PORTA_APERTA
    private func onEntraPortaAperta(motivo: String) {
        // Reset timeout
        gestoreTimeout?.reset()

        // Apri porta con motivo
        porta.apri(motivo: motivo)

        // Mostra messaggio successo
        display.mostraMessaggio("Accesso consentito - Porta aperta", tipo: .successo)

        Logger.info("✅ Porta aperta - Motivo: \(motivo)")

        // Programma chiusura automatica (15s)
        let timer = Timer.scheduledTimer(withTimeInterval: 15.0, repeats: false) { [weak self] _ in
            self?.porta.chiudi()

            // Torna a IDLE dopo animazione chiusura (1.5s)
            DispatchQueue.main.asyncAfter(deadline: .now() + 1.5) {
                self?.transizione(.idle)
            }
        }

        // Salva timer in porta per poterlo cancellare
        porta.timerChiusuraAutomatica = timer
    }

    /// Entra in stato TIMEOUT
    private func onEntraTimeout() {
        Logger.warning("⏱️ Timeout inattività raggiunto")

        // Mostra messaggio timeout
        display.mostraMessaggio("Timeout - Operazione annullata", tipo: .warning)

        // Reset e torna a IDLE dopo 2 secondi
        DispatchQueue.main.asyncAfter(deadline: .now() + 2.0) { [weak self] in
            self?.transizione(.idle)
        }
    }

    // MARK: - Azioni Pubbliche

    /// Inserisci moneta
    func inserisciMoneta(_ valore: Double) {
        // Se non in pagamento, inizia pagamento
        if stato == .idle {
            transizione(.pagamentoMonete)
        }

        guard stato == .pagamentoMonete else {
            Logger.warning("⚠️ Tentativo inserimento moneta in stato \(stato.rawValue)")
            return
        }

        // Inserisci moneta
        guard gettoniera.inserisciMoneta(valore) else {
            return
        }

        // Reset timeout (ogni moneta resetta il timeout)
        gestoreTimeout?.reset()
        gestoreTimeout?.avvia()

        // Aggiorna display con importo rimanente
        let rimanente = gettoniera.getImportoRimanente()
        display.mostraMessaggio("Inserisci monete", tipo: .info)
        display.mostraImporto(rimanente)

        // Verifica se pagamento completato
        verificaPagamento()
    }

    /// Verifica pagamento completato
    private func verificaPagamento() {
        guard stato == .pagamentoMonete else {
            return
        }

        if gettoniera.isPagamentoCompletato {
            Logger.info("✅ Pagamento completato")
            transizione(.portaAperta, dati: ["motivo": "monete"])
        }
    }

    /// Inizia pagamento con carta di credito
    func iniziaPagamentoCarta() {
        guard stato == .idle else {
            Logger.warning("⚠️ Tentativo pagamento carta in stato \(stato.rawValue)")
            return
        }

        transizione(.pagamentoCarta)
    }

    /// Elabora pagamento con carta
    func elaboraPagamentoCarta(numeroCarta: String) {
        guard stato == .pagamentoCarta else {
            return
        }

        display.mostraMessaggio("Elaborazione...", tipo: .info)

        // Simula elaborazione (500ms)
        DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) { [weak self] in
            guard let self = self else { return }

            // Verifica se carta VISA valida
            if Validatore.isCartaValida(numeroCarta) {
                Logger.info("✅ Pagamento carta accettato: \(numeroCarta)")
                self.display.mostraMessaggio("Pagamento accettato", tipo: .successo)

                DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) {
                    self.transizione(.portaAperta, dati: ["motivo": "carta"])
                }
            } else {
                Logger.warning("⚠️ Pagamento carta rifiutato: \(numeroCarta)")
                self.display.mostraMessaggio("Pagamento rifiutato - Solo VISA", tipo: .errore)

                DispatchQueue.main.asyncAfter(deadline: .now() + 2.0) {
                    self.transizione(.idle)
                }
            }
        }
    }

    /// Verifica QR code
    func verificaQR(codice: String) {
        guard stato == .idle else {
            Logger.warning("⚠️ Tentativo verifica QR in stato \(stato.rawValue)")
            return
        }

        transizione(.verificaQR, dati: ["codice": codice])
    }

    /// Verifica carta autorizzata
    func verificaCarta(codice: String) {
        guard stato == .idle else {
            Logger.warning("⚠️ Tentativo verifica carta in stato \(stato.rawValue)")
            return
        }

        transizione(.verificaCarta, dati: ["codice": codice])
    }

    /// Helper per verifica accesso con codice autorizzato (DRY helper)
    private func verificaAccessoConCodice(codice: String, tipoIngresso: String) {
        display.mostraMessaggio("Verifica in corso...", tipo: .info)

        // Simula verifica (500ms)
        DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) { [weak self] in
            guard let self = self else { return }

            let autorizzato = Validatore.isCodiceAutorizzato(codice)

            if autorizzato {
                Logger.info("✅ \(tipoIngresso) \(codice): AUTORIZZATO")
                self.display.mostraMessaggio("Accesso autorizzato", tipo: .successo)

                DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) {
                    self.transizione(.portaAperta, dati: ["motivo": tipoIngresso.lowercased()])
                }
            } else {
                Logger.warning("⚠️ \(tipoIngresso) \(codice): NON AUTORIZZATO")
                self.display.mostraMessaggio("Accesso negato", tipo: .errore)

                DispatchQueue.main.asyncAfter(deadline: .now() + 2.0) {
                    self.transizione(.idle)
                }
            }
        }
    }

    /// Handler passaggio persona (Feature 002)
    func onPassaggioPersona() {
        guard stato == .portaAperta else {
            Logger.warning("⚠️ Tentativo passaggio persona con porta non aperta - Ignorato")
            return
        }

        Logger.debug("🚶 Click pulsante \"Persona passata\"")

        // Mostra feedback su display
        display.mostraMessaggio("Passaggio rilevato - Porta in chiusura", tipo: .info)

        // Chiusura manuale porta (cancella timer e chiude)
        porta.chiudiManuale()

        // Torna a IDLE dopo animazione chiusura (1.5s)
        DispatchQueue.main.asyncAfter(deadline: .now() + 1.5) { [weak self] in
            self?.transizione(.idle)
        }
    }

    /// Reset completo del chiosco (per debugging)
    func reset() {
        Logger.warning("🔄 Reset manuale chiosco")
        transizione(.idle)
    }
}
